### PR TITLE
ham/eng 156 metadata api

### PR DESCRIPTION
**Pull request by ham-evans**
- metadata api and metadata json
- updating metadata

### CODE REVIEW

**Review by stormcloud266 — APPROVED** [review]
Everything is looking good to me

File: pages/api/metadata/ikachievement.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import metadata from "@nfts/metadata/IKAchievementMetadata.json";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const { tokenid } = req.query;
+
+  if (!tokenid || typeof tokenid === "object") return res.status(500).end();
```
Comment (illepic) [inline pages/api/metadata/ikachievement.ts]: Change this to !== string

File: pages/api/metadata/ikachievement.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  const tokenId = parseInt(tokenid, 10);
+
+  if (tokenId >= metadata.length) return res.status(500).end();
+
+  res.status(200).json(metadata[tokenId]);
```
Comment (illepic) [inline pages/api/metadata/ikachievement.ts]: Just `res.json(metadata[tokenId])`

**Review by illepic — CHANGES_REQUESTED** [review]

**Comment by illepic** [comment]
Also tsconfig conflict, easy fix.

**Review by illepic — APPROVED** [review]

**Pull request merged**